### ISSUE TITLE
0.0.76 regression: cached classmethod rejected as a generic instance attribute

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#4431]
### Summary

This standalone example passes with **ty 0.0.75** and fails with **0.0.76**.

```python
from functools import lru_cache
from typing import Generic, TypeVar

V = TypeVar('V')

class A(Generic[V]):
    @classmethod
    @lru_cache
    def f0(cls) -> 'A[V]':
        return cls()

A.f0()
```

Save as `repro.py` and run:

```sh
ty check --python-version 3.12 repro.py
```

**Expected:** successful checking. `f0` is a classmethod descriptor, and `A.f0()` works at runtime, returning a cached `A` instance.

**Actual on 0.0.76:** exit 1, with:

```text
error[invalid-attribute-access]: Cannot access generic instance attribute `f0` through a class
```

The diagnostic points to `A.f0()`.

Two controls pass with both versions:

- Remove `@lru_cache`, retaining `@classmethod`.
- Replace the `TypeVar`/`Generic` declaration with `class A[V]:`, retaining both decorators and the same method body.

The failure also reproduces with `@lru_cache(maxsize=128, typed=True)` and with a generic `NamedTuple` subclass.

The release includes [#28123](https://github.com/astral-sh/ruff/pull/28123), which adds generic-instance-attribute checks. Its stated intent is to preserve descriptor behavior. This looks related, but I have not bisected the regression.

Tested with the official macOS ARM64 binaries, Python 3.12 typing mode:

```text
ty 0.0.75 (a553dcc1a 2026-08-26)
ty 0.0.76 (1940c8a75 2026-08-31)
```

Pyrefly appears to have the same bug but other type checkers allow this.
